recipe/recipe.yaml

```diff
@@ -37,71 +33,97 @@ source:
+      # Since this meta package is noarch: generic, we only need to build it
+      # for one target platform (does not matter which one)
+      skip: target_platform != 'linux-64'
+      noarch: generic
+      string: h${{ hash }}_${{ polars_variant | replace('-', '_') }}_${{ build }}
+      variant:
+        down_prioritize_variant: ${{ 0 if polars_variant == 'default' else 1 }}
+    requirements:
+      run:
+        - ${{ 'polars-' ~ polars_variant }}==${{ version }}
+    tests:
+      - python:
+          pip_check: true
+          imports:
+            - polars
+          python_version: ${{ python_min }}.*
+      - script:
+          - python -c "from polars import DataFrame"
+  - package:
+      name: polars-${{ polars_variant }}
+    build:
+      skip: not (match(python, python_min ~ ".*") and is_abi3)
+      python:
+        version_independent: true
+    requirements:
+      build:
+        - if: build_platform != target_platform
+          then:
+            - python
+            - cross-python_${{ target_platform }}
+            - crossenv
+            - maturin >=1.3.2,<2
+        - ${{ compiler('c') }}
+        - ${{ compiler('cxx') }}
+        - ${{ stdlib('c') }}
+        - ${{ compiler('rust') }}
+        - if: build_platform == 'win-64'
+          then:
+            - posix
+        - cmake
+        - if: unix
+          then:
+            - make
+        - cargo-bundle-licenses
+        # TODO: remove once https://github.com/prefix-dev/rattler-build/issues/1000 is resolved
+        - if: target_platform == 'linux-64' or target_platform == 'osx-64'
+          then:
+            - sed
+        - if: target_platform == 'win-64'
+          then:
+            - m2-sed
+      host:
         - python
-        - cross-python_${{ target_platform }}
-        - crossenv
+        - python-abi3
+        - pip
         - maturin >=1.3.2,<2
-    - ${{ compiler('c') }}
-    - ${{ compiler('cxx') }}
-    - ${{ stdlib('c') }}
-    - ${{ compiler('rust') }}
-    - if: build_platform == 'win-64'
-      then:
-        - posix
-    - cmake
-    - if: unix
-      then:
-        - make
-    - cargo-bundle-licenses
-    # TODO: remove once https://github.com/prefix-dev/rattler-build/issues/1000 is resolved
-    - if: target_platform == 'linux-64' or target_platform == 'osx-64'
-      then:
-        - sed
-    - if: target_platform == 'win-64'
-      then:
-        - m2-sed
-  host:
-    - python
-    - python-abi3
-    - pip
-    - maturin >=1.3.2,<2
-  run:
-    - python
-    - numpy >=1.16.0
-    - if: python <3.11
-      then:
-        - typing_extensions >=4.0.0
-    - if: python >=3.10
-      then:
-        - packaging
-  ignore_run_exports:
-    from_package:
-      - cross-python_${{ target_platform }}
+      run:
+        - python
+        - numpy >=1.16.0
+        - if: python <3.11
+          then:
+            - typing_extensions >=4.0.0
+        - if: python >=3.10
+          then:
+            - packaging
```

Comment:
Unrelated to this PR really but I just stumbled upon that. @pavelzw do you know if this really works as intended? As we're only building one package for Python 3.9 now, I'd think that this evaluates in an unexpected way? https://conda-metadata-app.streamlit.app/?q=conda-forge%2Flinux-64%2Fpolars-1.29.0-py39h441220d_0.conda also looks weird; we only include `packaging` in the dependencies, even for Python 3.9 🤔

Reply:
ah right; this doesn't work anymore since our migration to abi3. we should make this an unconditional dependency in that case and just don't care that there is a dependency too many for python=3.9; it's EOL anyway soon

Reply:
We're also missing `typing_extensions` though, right? That'll still be applicable for a bit...

Reply:
i don't think they are needed at all actually, didn't find them in https://github.com/pola-rs/polars/blob/main/py-polars/pyproject.toml